apps/web/src/pages/Skills/UpdateUserSkillPage.tsx

```diff
@@ -0,0 +1,657 @@
+  skill: Skill;
+  experiences: Experience[];
+  userSkill?: Maybe<UserSkill>;
+}
+
+const UpdateUserSkillForm = ({
+  userId,
+  experiences,
+  skill,
+  userSkill,
+}: UpdateUserSkillFormProps) => {
+  const intl = useIntl();
+  const paths = useRoutes();
+  const navigate = useNavigate();
+  const [isFormDialogOpen, setFormDialogOpen] = React.useState<boolean>(false);
+  const [currentExperience, setCurrentExperience] =
+    React.useState<Experience | null>(null);
+  const skillName = getLocalizedName(skill.name, intl);
+  const skillDescription = getLocalizedName(skill.description, intl);
+  const hasUserSkill = notEmpty(userSkill);
+  const linkedExperiences = userSkill?.experiences?.filter(notEmpty);
+
+  const availableExperiences = experiences.filter(
+    (exp) =>
+      !linkedExperiences?.find(
+        (existingExperience) => existingExperience.id === exp.id,
+      ),
+  );
+
+  const isTechnical = skill.category === SkillCategory.Technical;
+  const levelGetter = isTechnical
+    ? getTechnicalSkillLevel
+    : getBehaviouralSkillLevel;
+  const levelDefinitionGetter = isTechnical
+    ? getTechnicalSkillLevelDefinition
+    : getBehaviouralSkillLevelDefinition;
+  const levelOptions = getSortedSkillLevels().map((skillLevel) => ({
+    value: skillLevel,
+    label: <strong>{intl.formatMessage(levelGetter(skillLevel))}</strong>,
+    contentBelow: (
+      <p data-h2-margin="base(x.15, 0, x.5, x1)">
+        {intl.formatMessage(levelDefinitionGetter(skillLevel))}
+      </p>
+    ),
+  }));
+
+  const [{ fetching: creating }, executeCreateMutation] =
+    useCreateUserSkillMutation();
+  const [{ fetching: updating }, executeUpdateMutation] =
+    useUpdateUserSkillMutation();
+  const [{ fetching: deleting }, executeDeleteMutation] =
+    useDeleteUserSkillMutation();
+  const mutating = creating || updating || deleting;
+
+  const handleSuccess = (msg?: React.ReactNode) => {
+    toast.success(
+      msg ||
+        intl.formatMessage({
+          defaultMessage: "Successfully updated skill!",
+          id: "Vfa3Ek",
+          description: "Message displayed when a user updates a skill",
+        }),
+    );
+    navigate(paths.skillLibrary());
+  };
+
+  const handleError = (msg?: React.ReactNode) => {
+    toast.error(
+      msg ||
+        intl.formatMessage({
+          defaultMessage: "Error: updating skill failed",
+          id: "CUxHd8",
+          description:
+            "Message displayed to user after skill fails to be updated.",
+        }),
+    );
+  };
+
+  const handleSubmit = (values: FormValues) => {
+    if (hasUserSkill) {
+      executeUpdateMutation({
+        id: userSkill.id,
+        userSkill: values,
+      })
+        .then(() => handleSuccess())
+        .catch(() => handleError());
+    } else {
+      executeCreateMutation({
+        userId,
+        skillId: skill.id,
+        userSkill: values,
+      })
+        .then(() => handleSuccess())
+        .catch(() => handleError());
```

Comment:
Should the successful creation not have a different message? You are claiming a skill and adding it to your profile which is distinct from updating your claimed skills.

Reply:
So, we never actually expect a user to "create" a user skill on this page. I guess there are two streams of thoughts by "creating" a user skill are you:

1. Creating a record of a claimed skill or,
2. Updating your currently claimed skills

I went with option 2 to reduce the number of strings we are translating (laziness). Also. no copy was supplied for the toasts and I couldn't think of language that made sense because creating isn't really what we are doing here 🤔 Maybe "Claimed skill successfully"? Or "Added skill to library"?

Reply:
Ah, well, I am not 100% sure about how all this is working either. I thought the action was claiming a skill that adds a skill entity to your profile

Reply:
Nope! This is the update page. While you technically can "create/claim" a skill here that is not the intention of this page. This is the page the edit a skill you have already claimed.

The actual creating is going to happen in https://github.com/GCTC-NTGC/gc-digital-talent/issues/6936

Reply:
Okay, thanks! That makes sense, then this comment can be disregarded